sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -304,6 +310,11 @@ abstract class JdbcDialect extends Serializable with Logging{
    */
   @Since("3.3.0")
   def compileAggregate(aggFunction: AggregateFunc): Option[String] = {
```

Comment:
I don't think we should add new APIs to jdbc dialect. Can we leverage `V2ExpressionBuilder` to make it easier for dialects to compile aggregates?